src/LinAlg/SparseMatrix.C

```diff
@@ -344,31 +344,45 @@ const Real& SparseMatrix::operator () (size_t r, size_t c) const
 
 void SparseMatrix::dump (std::ostream& os, LinAlg::StorageFormat format, const char* label)
 {
-  if (label) os << label <<" = [\n";
+  if (label && format != LinAlg::MATRIX_MARKET) os << label <<" = [\n";
+  std::string trail = ";\n";
```

Comment:
I did not see the need for introducing the `trail`variable here, except for slightly added complexity It is only for the new matrix format it is changed to '\n' such that the hard-coding ";\n" should still work fine for the Matlab format.

Reply:
it's supposed to be "];\n" - with the bracket. so it's a bug.

Reply:
Right, I missed there was no break at the end of that block. Then it is clear. (this case fall-through is sometimes hard to spot, and the newer compilers warns against it also, for the good or bad depending on taste).